lib/symboldatabase.cpp

```diff
@@ -7066,7 +7067,7 @@ void SymbolDatabase::setValueTypeInTokenList(bool reportDebugWarnings, Token *to
                         }
                     }
                 }
-                if (tok->astParent() && Token::Match(tok->astOperand1(), "%name%|::")) {
+                if (tok->astParent() && tok->astOperand1() && tok->astOperand1()->isCpp() && Token::Match(tok->astOperand1(), "%name%|::")) {
```

Comment:
Is it possible to use ` tok->astParent()->isCpp()`?

Reply:
It could possibly just be `tok->isCpp()` as it is global for all tokens from the same list.

I chose that since it was the underlying token we work from for the detection.

Reply:
I will adjust that. I missed some unnecessary calls.

Reply:
Done. There was an even cleaner solution.